source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -45,6 +45,9 @@ void setLambdaHeaders(Http::RequestHeaderMap& headers, const absl::optional<Arn>
                       InvocationMode mode) {
   headers.setMethod(Http::Headers::get().MethodValues.Post);
   headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", arn->arn()));
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.lambda_sanitize_host_header")) {
+    headers.setHost("lambda");
```

Comment:
please consider adding a small code comment indicating why this is ok. I am not able to think of a very good reason, but should give indication to the user that this change should be safe, but if is creating unwanted behavior then set the runtime flag to false as temp mitigation and surface the issue on Github.